inference-engine/src/gna_plugin/backend/dnn_components.cpp

```diff
@@ -9,31 +9,58 @@
 #include <legacy/ie_layers.h>
 #include <iomanip>
 #include "backend/dnn_types.h"
+#include <details/caseless.hpp>
+#include <layers/gna_copy_layer.hpp>
 
 #include "dnn_components.hpp"
 
 using namespace GNAPluginNS;
+using namespace GNAPluginNS::backend;
 
-intel_dnn_component_t & backend::DnnComponents::addComponent(const std::string layerName, const std::string layerMetaType) {
-    components.emplace_back(layerName, intel_dnn_component_t());
-    auto &currentComponent = components.back().second;
-#ifdef PLOT
-    currentComponent.original_layer_name = components.back().first.c_str();
-    std::cout << "IR layer : " << std::left << std::setw(20) << layerName << " " << layerMetaType << "_" << components.size() - 1 << std::endl;
+intel_dnn_component_t & DnnComponents::addComponent(const std::string layerName, const std::string layerMetaType) {
+    auto isDelayed = InferenceEngine::details::CaselessEq<std::string>()(layerMetaType, DelayedCopyLayerName);
+    delayedOperations += isDelayed ? 1 : 0;
+    components.emplace_back(DnnComponentExtra{layerName, {}, isDelayed});
+    auto &currentComponent = components.back().dnnComponent;
+#if defined(PLOT) || defined(GEN_STATS)
+    currentComponent.original_layer_name = components.back().name.c_str();
 #endif
+    int execOrder = 0;
+    if (!isDelayed) {
+        execOrder = components.size() - 1 - delayedOperations;
+    } else {
+        // todo: not perfect - propose to create mapping table that will be printed out by extra request
+        execOrder = -delayedOperations;
```

Comment:
This cause compilation error in Visual studio compiler.

Reply:
i suggest to print execorder in initGNAstruct - it allows correct ordering and easier to find in log messages